src/main/java/com/aws/greengrass/deployment/activator/DeploymentActivator.java

```diff
@@ -138,4 +138,42 @@ protected UpdateBehaviorTree createDeploymentMergeBehavior(long deploymentTimest
 
         return rootMergeBehavior;
     }
+
+    private UpdateBehaviorTree createRollbackMergeBehavior() {
+        // root: MERGE
+        //   services: MERGE
```

Comment:
we should be removing any services that were removed right?

Reply:
Default should be replace when we rollback. What are the few (if any) instances where we actually do want to keep updated/added values?

Reply:
That should be taken care of through the service dependency tree and `service.remove()` that happens as part of `removeObsoleteServices` part of roll-forward/rollback deployment.

Reply:
I could make it replace for services, because I also agree that most things for rollbacks should be replace and only a few things like the private and runtime namespaces should be overridden to merge. But I've left it as 'merge' behavior to be careful

Reply:
question: Do we lack test coverage to make you more confident in making it replace? What is better about merge makes you more confident in that behavior?

Reply:
At first it was mostly about causing least disruption and because I know that leaving this as merge doesn't mean we don't remove any service instances that must be removed. Another reason that I just realized - the order for rollback steps where you merge config then shut down and close obsolete services, and then remove the configuration for those services that we currently already follow is more appropriate. If we made everything 'replace' here then we remove config for services before closing them and it could lead to situations like no longer having the shutdown script for a service and not cleaning it up properly.

Reply:
thanks for diving into that. That makes sense